tests/unit/test_charm_state.py

```diff
@@ -104,7 +116,102 @@ def test_from_charm_arch(monkeypatch: pytest.MonkeyPatch, arch: str, expected_ar
     monkeypatch.setattr(platform, "machine", mock_machine)
     mock_charm = MagicMock(spec=GithubRunnerCharm)
     mock_charm.config = {}
+    mock_charm.model.relations = defaultdict(lambda: None)
 
     state = State.from_charm(mock_charm)
 
     assert state.arch == expected_arch
+
+
+def test_ssh_debug_info_from_charm_no_relations():
+    """
+    arrange: given a mocked charm that has no ssh-debug relations.
+    act: when SSHDebug.from_charm is called.
+    assert: None is returned.
+    """
+    mock_charm = MagicMock(spec=GithubRunnerCharm)
+    mock_charm.model.relations = {DEBUG_SSH_INTEGRATION_NAME: []}
+
+    assert SSHDebugInfo.from_charm(mock_charm) is None
+
+
+@pytest.mark.parametrize(
+    "invalid_relation_data",
+    [
+        pytest.param(
+            {
+                "host": "invalidip",
+                "port": "8080",
+                "rsa_fingerprint": "fingerprint_data",
+                "ed25519_fingerprint": "fingerprint_data",
+            },
+            id="invalid host IP",
+        ),
+        pytest.param(
+            {
+                "host": "127.0.0.1",
+                "port": "invalidport",
+                "rsa_fingerprint": "fingerprint_data",
+                "ed25519_fingerprint": "fingerprint_data",
+            },
+            id="invalid port",
```

Comment:
What about invalid fingerprint? Like it is an integer or something?

Reply:
Valid point, let me add a regex for fingerprint `SHA:<fingerprint>` and test for those as well. Thank you!